isis/tests/FunctionalTestsCsminit.cpp

```diff
@@ -0,0 +1,218 @@
+#include "csminit.h"
+#include "spiceinit.h"
+
+#include <QString>
+#include <iostream>
+
+#include "TestCsmPlugin.h"
+#include "Fixtures.h"
+#include "TestUtilities.h"
+#include "StringBlob.h"
+#include "FileName.h"
+
+#include <nlohmann/json.hpp>
+using json = nlohmann::json;
+
+#include "gmock/gmock.h"
+
+using namespace Isis;
+
+static QString APP_XML = FileName("$ISISROOT/bin/xml/csminit.xml").expanded();
+
+class CSMPluginFixture : public TempTestingFiles {
+  protected:
+      TestCsmPlugin *plugin;
```

Comment:
The plugin doesn't need to be a part of the fixture. It should automatically self register.

Reply:
Thanks! I was having trouble getting this to work initially, but it works great now!